esmvalcore/preprocessor/_regrid.py

```diff
@@ -639,6 +639,62 @@ def _load_generic_scheme(scheme: dict):
     return loaded_scheme
 
 
+_CACHED_REGRIDDERS: dict[tuple, dict] = {}
+
+
+def _get_regridder(src_cube: Cube, tgt_cube: Cube, scheme: str | dict):
+    """Get regridder to actually perform regridding.
+
+    Note
+    ----
+    If possible, this uses an existing regridder to reduce runtime (see also
+    https://scitools-iris.readthedocs.io/en/latest/userguide/
+    interpolation_and_regridding.html#caching-a-regridder.)
+
+    """
+    # Use existing regridder if possible. This first checks the regridding
+    # scheme name and shapes of source and target coordinates, and only if
+    # these match check coordinates themselves (this is much more expensive).
+    coord_key = _get_coord_key(src_cube, tgt_cube)
```

Comment:
have completely dropped the option that regrid preprocessor be optionally called with an "MxN" target grid rather than a target grid cube? If we did, carry on - if we didn't, then this will break it

Reply:
or this one gets used only when target grid is a cube?

Reply:
At this point in the code, `tgt_cube` will always be a cube, since it has been create from `_get_target_grid_cube`. It's still possible to use `MxN` specs of course.

Reply:
gotcha, thanks, Manu (I was suspecting that's the case, but better confirm, like in aviation :airplane: )